docs/source/getting_started/main_concepts.md

```diff
@@ -0,0 +1,124 @@
+
+### Start of training
+StreamingDataset downloads the `index.json` files for input streams, and uses the information they contain to partition samples across nodes, ranks, and workers. The diagram below shows this process:
+
+<img src="../_static/images/streaming_partitioning.png" alt="MDS Writing" width="800"/>
+
+Let's understand what's happening here.
+
+#### Remote data streams
+The `index.json` files and shards for multiple streams are stored in the cloud. A Stream typically corresponds to one data source. The overall dataset combines samples from all input streams. Two streams are shown here, but StreamingDataset supports combining any number. Streams can be mixed in various [ways](../dataset_configuration/mixing_data_sources.md), making final dataset composition flexible.
+
+<img src="../_static/images/remote_streams.png" alt="Remote Streams" width="800"/>
+
+Below, we pass in a list of {class}`streaming.Stream` objects to a {class}`streaming.StreamingDataset`, and also specify the proportion of the overall dataset we want to take from each stream.
+<!--pytest.mark.skip-->
+```python
+# Stream 1 uses its own set of shard files and will be 1/4 of the training dataset.
+stream_1 = Stream(
+    remote='s3://stream_1/directory',
+    local='/local/cache/stream_1',
+    batch_size=4,
+    proportion=0.25
+)
+# Stream 2 is similar to above, but will be 3/4 of the training dataset.
+stream_2 = Stream(
+    remote='s3://stream_2/directory',
+    local='/local/cache/stream_2',
+    batch_size=4,
+    proportion=0.75
+)
+
+# This dataset uses multiple streams.
+dataset = StreamingDataset(streams=[stream_1, stream_2], batch_size=4)
+```
+
+If using a single stream, we could just specify the `remote` and `local` locations directly instead:
+<!--pytest.mark.skip-->
+```python
+dataset = StreamingDataset(remote='s3://some/path', local='/local/path', batch_size=4)
+```
+
+#### Sample partitioning
+Dataset samples are partitioned among nodes, GPUs, and workers. This partitioning reduces the redundant downloads during training, making it much more performant.
+
+<img src="../_static/images/sample_partition.png" alt="Sample Partition" width="800"/>
+
+In the diagram above, we have 2 nodes, 8 GPUs per node, and 2 dataloader workers per GPU. These values can vary depending on the training configuration. The number of nodes and ranks is detected through PyTorch, and the number of workers is passed in through the DataLoader. Zooming into the sample partition for just one GPU, the samples are split up between dataloader workers (2 per GPU above) and grouped together by GPU batch size (4 above).
+
+#### Dataset shuffling
+You can shuffle the samples within each node using one of our specialized [shuffling algorithms](../dataset_configuration/shuffling.md#shuffling-algorithms). Having a shuffled dataset is highly important for ML model training. We shuffle samples within each node to ensure that overall download demand is minimized.
```

Comment:
Ah it's not that shuffling itself helps decrease download demand, it's that shuffling within each node, rather than across nodes, helps decrease download demand. Will expand on this part.

Reply:
edited